cpp/include/cugraph/graph_view.hpp

```diff
@@ -958,6 +979,18 @@ class graph_view_t<vertex_t,
     return vertex_partition_view_t<vertex_t, false>(this->number_of_vertices());
   }
 
+  void attach_mask_view(graph_mask_view_t<vertex_t, edge_t>& mask)
+  {
+    mask_ = std::make_optional<graph_mask_view_t<vertex_t, edge_t>*>(&mask);
+  }
+
+  std::optional<graph_mask_view_t<vertex_t, edge_t>> get_mask_view(size_t partition_idx = 0) const
```

Comment:
This should not take `partition_idx`. We'd better not use one graph_mask_view_t object per edge partition; note that graph_mask_view_t object has vertex mask as well. In case of MG, a graph_mask_view_t object may have more than one edge mask arrays internally.